src/opera/parser/tosca/csar.py

```diff
@@ -1,31 +1,286 @@
 import shutil
+from abc import ABC, abstractmethod
+from pathlib import Path, PurePath
+from tempfile import mktemp
+from typing import List, IO, Optional
+from zipfile import ZipFile
+
 import yaml
 
-from pathlib import Path
-from tempfile import TemporaryDirectory, mktemp
-from zipfile import ZipFile
+from opera.error import ParseError, OperaError
+from opera.utils import determine_archive_format
+
+
+class CsarMeta:
+    SUPPORTED_META_FILE_VERSIONS = {"1.1"}
+    SUPPORTED_CSAR_VERSIONS = {"1.1"}
+    METADATA_PATH = "TOSCA-Metadata/TOSCA.meta"
+
+    KEY_NAME = "Name"
+    KEY_CONTENT_TYPE = "Content-Type"
+    KEY_TOSCA_META_FILE_VERSION = "TOSCA-Meta-File-Version"
+    KEY_CSAR_VERSION = "CSAR-Version"
+    KEY_CREATED_BY = "Created-By"
+    KEY_ENTRY_DEFINITIONS = "Entry-Definitions"
+
+    REQUIRED_KEYS = {KEY_TOSCA_META_FILE_VERSION, KEY_CSAR_VERSION, KEY_CREATED_BY}
```

Comment:
IDE says: `PEP 8: E501 line too long (83 > 79 characters).` Sir, please fix all the lines that are too long to be consistent with other opera Python files.

Reply:
I'll wrap everything at 79, but I'd like to expres interest to change this to something more humane, such as 120.

Reply:
Yes I agree, wrapping lines to 79 chars gives too little space for coding without worrying that you have to do a new line every few minutes or so. I've created a new issue for that: #163.

Reply:
Line 118 in the `csar.py` still exceeds 79. Now that's really annoying. If you want you can fix this limit to 120 before me in a separate PR.